From: Ricardo Neri <ricardo.neri@ti.com>
Date: Tue, 29 Mar 2011 00:25:51 +0000 (-0600)
Subject: OMAP4: HDMI: Disable HDMI mode if sink does not have HDMI ID
X-Git-Tag: android-2.6.35-2.3-omap4.13.1~39
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=6c29a91b83c638db25d7ff4d132224367c3ae37e

OMAP4: HDMI: Disable HDMI mode if sink does not have HDMI ID

Examine EDID for the HDMI IEEE unique identifier in the Vendor-Specific
Data Block. If present, enable HDMI mode in HDMI IP core; if absent
disable HDMI mode (DVI mode).

Change-Id: I32b38b15d2e5c94fe091e9b96df9fa617c2d57d7
Signed-off-by: Ricardo Neri <ricardo.neri@ti.com>
---

diff --git a/drivers/video/omap2/dss/hdmi.c b/drivers/video/omap2/dss/hdmi.c
index 9d28704..4a70d23 100644
--- a/drivers/video/omap2/dss/hdmi.c
+++ b/drivers/video/omap2/dss/hdmi.c
@@ -1167,7 +1167,7 @@ static int hdmi_power_on(struct omap_dss_device *dssdev)
 		hdmi.cfg.vsi_enabled = false;
 	}
 
-	hdmi.cfg.hdmi_dvi = hdmi.mode;
+	hdmi.cfg.hdmi_dvi = hdmi_has_ieee_id((u8 *)edid) && hdmi.mode;
 	hdmi.cfg.video_format = hdmi.code;
 	hdmi.cfg.supports_ai = hdmi_ai_supported(edid);
 
@@ -1245,7 +1245,7 @@ static int hdmi_min_enable(void)
 		hdmi.cfg.vsi_enabled = false;
 	}
 
-	hdmi.cfg.hdmi_dvi = hdmi.mode;
+	hdmi.cfg.hdmi_dvi = hdmi_has_ieee_id((u8 *)edid) && hdmi.mode;
 	hdmi.cfg.video_format = hdmi.code;
 	hdmi_lib_enable(&hdmi.cfg);
 	return 0;